backend/src/bundles/avatar-videos/avatar-videos.service.ts

```diff
@@ -0,0 +1,133 @@
+    private videoService: VideoService;
+
+    public constructor(
+        azureAIService: AzureAIService,
+        fileService: FileService,
+        videoService: VideoService,
+    ) {
+        this.azureAIService = azureAIService;
+        this.fileService = fileService;
+        this.videoService = videoService;
+    }
+
+    private async saveAvatarVideo(url: string, id: string): Promise<string> {
+        const buffer = await this.azureAIService.getAvatarVideoBuffer(url);
+
+        const fileName = getFileName(id);
+
+        await this.fileService.uploadFile(buffer, fileName);
+        return this.fileService.getCloudFrontFileUrl(fileName);
+    }
+
+    public removeAvatarVideo(id: string): Promise<unknown> {
+        return this.azureAIService.removeAvatarVideo(id);
+    }
+
+    public async renderAvatarVideo(
+        payload: RenderAvatarVideoRequestDto & { userId: string },
+    ): Promise<RenderAvatarResponseDto> {
+        const { userId, ...avatarConfig } = payload;
+        const response = await this.azureAIService.renderAvatarVideo({
+            id: uuidv4(),
+            payload: avatarConfig,
+        });
+
+        this.checkAvatarProcessing(response.id, userId);
+
+        return response;
+    }
+
+    private checkAvatarProcessing(id: string, userId: string): void {
+        const interval = setInterval((): void => {
+            this.azureAIService
+                .getAvatarVideo(id)
+                .then((response) => {
+                    if (
+                        response.status ===
+                        GenerateAvatarResponseStatus.SUCCEEDED
+                    ) {
+                        this.handleSuccessfulAvatarGeneration({
+                            id,
+                            userId,
+                            url: response.outputs.result,
+                        })
+                            .then(() => {
+                                // TODO: NOTIFY USER
+                            })
+                            .catch((error) => {
+                                throw new HttpError({
+                                    message: error.message,
+                                    status: error.status,
+                                });
+                            })
+                            .finally(() => {
+                                clearInterval(interval);
+                            });
+                    } else if (
+                        response.status === GenerateAvatarResponseStatus.FAILED
+                    ) {
+                        // TODO: NOTIFY USER
+                        clearInterval(interval);
+                    }
+                })
+                .catch((error) => {
+                    clearInterval(interval);
+                    throw new HttpError({
+                        message: error.message,
+                        status: error.status,
+                    });
+                });
```

Comment:
Maybe, we can create a separate function for that and use async await so it will be more readable?